### PR TITLE
[YUNIKORN-2830] Update Kubernetes compile dependency to 1.31.0

### PR DESCRIPTION
**Pull request by craigcondit**
### What is this PR for?
Update our compile-time dependency on Kubernetes to v1.31.0. Also ensure that we are instantiating all required informers for scheduler plugins (there were several missing).

### What type of PR is it?
* [ ] - Bug Fix
* [x] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2830

### How should this be tested?
Existing e2e tests should pass.

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -18,186 +18,193 @@
...
+	sigs.k8s.io/kustomize/kyaml v0.17.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
-	sigs.k8s.io/yaml v1.3.0 // indirect
+	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
 replace (
-	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.12
+	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.13
 	github.com/petermattis/goid => github.com/petermattis/goid v0.0.0-20240327183114-c42a807a84ba
```
Comment (wilfred-s) [inline go.mod:172]: we should update to the latest go-deadlock 0.3.5 to get the up to date version of this.

File: go.mod [hunk 2]
```diff
@@ -18,186 +18,193 @@
...
-	k8s.io/component-helpers => k8s.io/component-helpers v0.29.6
-	k8s.io/controller-manager => k8s.io/controller-manager v0.29.6
-	k8s.io/cri-api => k8s.io/cri-api v0.29.6
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.29.6
-	k8s.io/dynamic-resource-allocation => k8s.io/dynamic-resource-allocation v0.29.6
-	k8s.io/endpointslice => k8s.io/endpointslice v0.29.6
+	golang.org/x/net => golang.org/x/net v0.26.0
+	golang.org/x/sys => golang.org/x/sys v0.21.0
+	golang.org/x/text => golang.org/x/text v0.16.0
+	golang.org/x/tools => golang.org/x/tools v0.21.1-0.20240508182429-e35e4ccd0d2d
```
Comment (wilfred-s) [inline go.mod:178]: Can we move to the latest of all of these:
```
	golang.org/x/crypto => golang.org/x/crypto v0.26.0
	golang.org/x/net => golang.org/x/net v0.28.0
	golang.org/x/sys => golang.org/x/sys v0.24.0
	golang.org/x/text => golang.org/x/text v0.17.0
	golang.org/x/tools => golang.org/x/tools v0.24.0
```
I still think we need to keep them in their own replace outside of the K8s dependencies

File: go.mod [hunk 3]
```diff
@@ -18,186 +18,193 @@
...
+	google.golang.org/protobuf => google.golang.org/protobuf v1.34.2
```
Comment (wilfred-s) [inline go.mod:179]: We need to update the scheduler-interface protobuf version and pull that one in then we should not need this one. K8s has already updated this in 1.31 Updates need to flow from SI -> core -> k8shim

**Review by wilfred-s — CHANGES_REQUESTED** [review]

File: go.mod [hunk 4]
```diff
@@ -18,186 +18,193 @@
...
+	sigs.k8s.io/kustomize/api v0.17.2 // indirect
+	sigs.k8s.io/kustomize/kyaml v0.17.1 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
-	sigs.k8s.io/yaml v1.3.0 // indirect
+	sigs.k8s.io/yaml v1.4.0 // indirect
 )
 
 replace (
-	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.12
+	github.com/opencontainers/runc => github.com/opencontainers/runc v1.1.13
```
[the remark this answers is not in the document]
Comment (craigcondit) [reply go.mod]: Done.

File: go.mod (hunk 1 above)
Comment (craigcondit) [reply go.mod]: Done.

File: go.mod (hunk 2 above)
Comment (craigcondit) [reply go.mod]: Done.

File: go.mod (hunk 3 above)
Comment (craigcondit) [reply go.mod]: SI ref updated, core PR is submitted.

**Review by pbacsko — APPROVED** [review]
LGTM

**Pull request closed without merging**